plugins/inputs/win_eventlog/win_eventlog.go

```diff
@@ -0,0 +1,515 @@
+	for _, pattern := range w.EventTags {
+		if matched, _ := filepath.Match(pattern, field); matched {
+			// Tags are not excluded
+			return true, "tags"
+		}
+	}
+
+	for _, pattern := range w.EventFields {
+		if matched, _ := filepath.Match(pattern, field); matched {
+			if w.shouldExclude(field) {
+				return false, "excluded"
+			}
+			return true, "fields"
+		}
+	}
+	return false, "excluded"
+}
+
+func (w *WinEventLog) shouldExcludeEmptyField(field string, fieldType string, fieldValue interface{}) (should bool) {
+	for _, pattern := range w.ExcludeEmpty {
+		if matched, _ := filepath.Match(pattern, field); matched {
+			switch fieldType {
+			case "string":
+				return len(fieldValue.(string)) < 1
+			case "int":
+				return fieldValue.(int) == 0
+			case "uint32":
+				return fieldValue.(uint32) == 0
+			}
+		}
+	}
+	return false
+}
+
+func (w *WinEventLog) evtSubscribe(logName, xquery string) (EvtHandle, error) {
+	var logNamePtr, xqueryPtr *uint16
+
+	sigEvent, err := windows.CreateEvent(nil, 0, 0, nil)
+	if err != nil {
+		return 0, err
+	}
+	defer windows.CloseHandle(sigEvent)
+
+	logNamePtr, err = syscall.UTF16PtrFromString(logName)
+	if err != nil {
+		return 0, err
+	}
+
+	xqueryPtr, err = syscall.UTF16PtrFromString(xquery)
+	if err != nil {
+		return 0, err
+	}
+
+	subsHandle, err := _EvtSubscribe(0, uintptr(sigEvent), logNamePtr, xqueryPtr,
+		0, 0, 0, EvtSubscribeToFutureEvents)
+	if err != nil {
+		return 0, err
+	}
+
+	return subsHandle, nil
+}
+
+func (w *WinEventLog) fetchEventHandles(subsHandle EvtHandle) ([]EvtHandle, error) {
+	var eventsNumber uint32
+	var evtReturned uint32
+
+	eventsNumber = 5
+
+	eventHandles := make([]EvtHandle, eventsNumber)
+
+	err := _EvtNext(subsHandle, eventsNumber, &eventHandles[0], 0, 0, &evtReturned)
+	if err != nil {
+		if err == ERROR_INVALID_OPERATION && evtReturned == 0 {
+			return nil, ERROR_NO_MORE_ITEMS
+		}
+		return nil, err
+	}
+
+	return eventHandles[:evtReturned], nil
+}
+
+func (w *WinEventLog) fetchEvents(subsHandle EvtHandle) ([]Event, error) {
+	var events []Event
+
+	eventHandles, err := w.fetchEventHandles(subsHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, eventHandle := range eventHandles {
+		if eventHandle != 0 {
+			event, err := w.renderEvent(eventHandle)
+			if err == nil {
+				// w.Log.Debugf("Got event: %v", event)
+				events = append(events, event)
+			}
+		}
+	}
+
+	for i := 0; i < len(eventHandles); i++ {
+		err := _EvtClose(eventHandles[i])
+		if err != nil {
+			return events, err
+		}
+	}
+	return events, nil
+}
+
+func (w *WinEventLog) renderEvent(eventHandle EvtHandle) (Event, error) {
+	var bufferUsed, propertyCount uint32
+
+	event := Event{}
+	err := _EvtRender(0, eventHandle, EvtRenderEventXml, uint32(len(w.buf)), &w.buf[0], &bufferUsed, &propertyCount)
+	if err != nil {
+		return event, err
+	}
+
+	eventXML, err := DecodeUTF16(w.buf[:bufferUsed])
+	if err != nil {
+		return event, err
+	}
+	xml.Unmarshal([]byte(eventXML), &event)
```

Comment:
It returns `err`. It would be safer to handle it.
(more over you don't handle error in `util_test.go` twice: in `writer.Write(word)` and `xml.Unmarshal([]byte(xmldata), &container)` - but these are in tests)